dot/build_spec_test.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package dot
+
+import (
+	"errors"
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildSpec_ToJSON(t *testing.T) {
+	tests := []struct {
+		name    string
+		genesis *genesis.Genesis
+		want    string
+		err     error
+	}{
+		{
+			name:    "name test",
+			genesis: &genesis.Genesis{Name: "test"},
+			want: `{
+    "name": "test",
```

Comment:
> nit: add a strings.TrimSpace(tt.want), .. when you do your assert.Equal and you can make the indenting look better on the expected JSON.

But doesn't `strings.TrimSpace` just trim the space at the beginning and end of the string?

> I've also seen people just populate the struct and `json.Marshal` the struct before asserting equal.

Yes and no since this can be risky if i.e. a field gets unexported. Better to assert with raw strings I'd say.

Reply:
Updated.